src/models/kernels.cu

```diff
@@ -107,5 +121,257 @@ void LaunchInt32ToInt64(const int32_t* src, int64_t* dst, int count, cudaStream_
   ConvertInt32ToInt64<<<num_blocks, block_size, 0, stream>>>(src, dst, count);
 }
 
+// Support head_size up to 128
+constexpr unsigned int kTileSize = 32;
+constexpr unsigned int kSeqTileSize = 16;
```

Comment:
Can we move these constants inside `ReorderPastStatesKernelLauncher` and pass them to `ReorderPastStatesKernel`?

Reply:
Seems like it must be `constexpr` within the context of this function since there is a memory allocation in the kernel:

```cpp
__shared__ float4 tile[kSeqTileSize][kTileSize + 1];
```

I'll put these in an unnamed namespace for now. Or if we want to, we can also create the \_\_shared\_\_ allocation outside of the kernel and define it as extern. Let me know if you have a preference.

Reply:
I ended up putting the constexprs in a struct of its own. This way, we continue benefitting from compile time memory allocation as opposed to dynamic memory allocation and also make the consts <somewhat> specific to the `ReorderPastStates` kernel